runtime/src/lib.rs

```diff
@@ -277,6 +277,12 @@ pub fn handle_mailbox_commands(drivers: &mut Drivers) -> CaliptraResult<()> {
 
         let cmd_ready = drivers.mbox.is_cmd_ready();
         if cmd_ready {
+            // We are about to handle a mailbox command; clear the sticky
+            // register so we can go back to sleep later.
+            if cfg!(feature = "fpga_realtime") {
+                drivers.soc_ifc.clear_mbox_notif_status();
+            }
```

Comment:
nit: We could probably clear this inside of `enter_idle` right after `mpmc_halt_and_enable_interrupts`, right? Just thinking it consolidates the interrupt logic a bit.

Also thinking of a case there some interrupt is fired but is_cmd_ready is false (although that's probably a bug if that happens since we only enable one interrupt..)

Reply:
No, because enter_idle doesn't know for sure that you're going to process the mailbox.

If we clear the interrupt status bit, but we don't get into this branch (for example, due to a race condition or unexpected interrupt), then we risk going to sleep forever, which would be very bad.